### PR TITLE
Add tests showing the limitations of file-local type shadowing

### PR DESCRIPTION
**Pull request by RikkiGibson**
Runtime has started to adopt file-local types in dotnet/runtime#71765. We had some questions come up about how the file type is supposed to be bound when other types with the same name are in scope. Adding a few tests to clarify the behavior, and to show in these specific cases that the behavior may be the same whether the 'file' modifier is used or not.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs [hunk 1]
```diff
@@ -1424,6 +1424,98 @@ void verify()
...
+            }
+
+            {{fileModifier}}class C
+            {
+                public static void M() => Console.WriteLine("OtherFile.cs");
+            }
+            """;
+
+        var verifier = CompileAndVerify(new[] { userCode, generatedCode }, expectedOutput: "OtherFile.cs\r\nProgram.cs");
+        verifier.VerifyDiagnostics();
```
[the remark this answers is not in the document]
Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs:1516]: Expanding 'Duplication_01' to show the behavior with and without a global qualifier.

**Review by cston — APPROVED** [review]

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs [hunk 2]
```diff
@@ -1424,6 +1425,98 @@ void verify()
...
+    }
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs:1518]: Consider adding a test variant where the file-type C is inside the proper namespace in source-generated code (that's another mitigation).

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2) with a minor suggestion

**Review by jaredpar — APPROVED** [review]

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs (hunk 2 above)
Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs:1518]: Done

**Pull request merged**